### PR TITLE
66 Create contact point schema.

### PR DESCRIPTION
**Pull request by akshayb2022**

### CODE REVIEW

File: ckanext/datitrentinoit/model/mapping.py [hunk 1]
```diff
@@ -172,6 +172,16 @@ def create_pro_package_dict(guid, swpentry: StatWebProEntry, metadata: StatWebMe
...
     extras['source_url'] = swpentry.get_url()
+    extras['creator'] = [{
+        "creator_identifier": "XGT4IE",
+        "creator_name": "ISPAT Istituto di statistica della provincia di Trento"
+    }]
+    extras['contact_point_name'] = "XGT4IE"
+    extras['contact_point_identifier'] = "ISPAT Istituto di statistica della provincia di Trento"
+    extras['contact_point'] = [{
+        "contact_point_identifier": "XGT4IE",
+        "contact_point_name": "ISPAT Istituto di statistica della provincia di Trento"
```
Comment (etj) [inline ckanext/datitrentinoit/model/mapping.py:183]: We could use a couple of costants for name and identifier ;)

Comment (etj) [inline ckanext/datitrentinoit/model/mapping.py:183]: Do we need contact point info to be replicated both in the extras and as a list?
By the doc contact point is 0..n, so the list should be the correct encoding.

File: ckanext/datitrentinoit/profile.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    """
+    Customization of DCATAP-IT for Regione Umbria
+    """
+
+    def parse_dataset(self, dataset_dict, dataset_ref):
+        return dataset_dict
+
+    def graph_from_dataset(self, dataset_dict, dataset_ref):
+        self._replace_poc_with_publisher(dataset_dict, dataset_ref)
+        self._augment_publisher_info(dataset_dict, dataset_ref)
```
Comment (etj) [inline ckanext/datitrentinoit/profile.py:27]: `graph_from_dataset()` is the needed extension point. The source code you are copying from is doing a couple of things, which are in the name of methods called.
I guess we only need a single method there that will replace the poc only (maybe a `replace_poc` may be a name informative enough). The current name is misleading.

File: ckanext/datitrentinoit/profile.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    def _augment_publisher_info(self, dataset_dict, dataset_ref):
+        publisher = self.g.value(subject=dataset_ref, predicate=DCT.publisher, object=None, any=False)
+
+        if publisher:
+            for key, predicate, otype in (
+                    ('contact_point_name', FOAF.mbox, Literal),
+                    ('contact_point_identifier', FOAF.homepage, URIRef),
+                    ('contact_point', FOAF.mbox, Literal)
+            ):
+                self.g.add((publisher, predicate, otype(self._get_dict_value(dataset_dict, key, 'N/A'))))
```
Comment (etj) [inline ckanext/datitrentinoit/profile.py:63]: We dont need it

File: ckanext/datitrentinoit/profile.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        # Remove previous poc
+        for s, p, o in g.triples((dataset_ref, DCAT.contactPoint, None)):
+            log.info(f"UmbriaProfile: Removing contactPoint {o}")
+            g.remove((s, p, o))
+            remove_unused_object(g, o, "contactPoint (UMBRIA)")
+
+        # Add new poc
+        pub_name = dataset_dict.get('contact_point_name')
+        pub_identifier = dataset_dict.get('contact_point_identifier')
+        create_point = dataset_dict.get('contact_point')
```
Comment (etj) [inline ckanext/datitrentinoit/profile.py:42]: ?

**Review by etj — CHANGES_REQUESTED** [review]
The use of `contact_point_name` `contact_point_identifier` and `contact_point` is confusing and overlapping.

I guess we need only the `contact_point` containing a list of  dict with keys `contact_point_name` `contact_point_identifier`.

File: ckanext/datitrentinoit/profile.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        org_dict = _get_org(dataset_dict.get('owner_org'))
+
+        # Remove previous poc
+        for s, p, o in g.triples((dataset_ref, DCAT.contactPoint, None)):
+            log.info(f"UmbriaProfile: Removing contactPoint {o}")
+            g.remove((s, p, o))
+            remove_unused_object(g, o, "contactPoint (UMBRIA)")
+
+        # Add new poc
+        create_point = dataset_dict.get('contact_point')
```
Comment (etj) [inline ckanext/datitrentinoit/profile.py:36]: @akshayb2022 why is this var called "create_point"?

File: ckanext/datitrentinoit/profile.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    """
+    Customization of DCATAP-IT for Regione Umbria
+    """
+
+    def parse_dataset(self, dataset_dict, dataset_ref):
+        return dataset_dict
+
+    def graph_from_dataset(self, dataset_dict, dataset_ref):
+        g = self.g
+        org_dict = _get_org(dataset_dict.get('owner_org'))
```
Comment (etj) [inline ckanext/datitrentinoit/profile.py:27]: What do you need this info for? Is it used?

**Review by etj — CHANGES_REQUESTED** [review]
Check review comments

File: ckanext/datitrentinoit/profile.py (hunk 6 above)
Comment (akshayb2022) [reply ckanext/datitrentinoit/profile.py]: Done

File: ckanext/datitrentinoit/profile.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        # Add new poc
+        contact_point = dataset_dict.get('contact_point')
+
+        poc = BNode()
+        g.add((dataset_ref, DCAT.contactPoint, poc))
+        g.add((poc, RDF.type, DCATAPIT.Organization))
+        g.add((poc, RDF.type, VCARD.Kind))
+        g.add((poc, RDF.type, VCARD.Organization))
+
+        g.add((poc, VCARD.fn, Literal(contact_point)))
```
Comment (etj) [inline ckanext/datitrentinoit/profile.py:34]: Contact point is stored as a json field, so it should be unpacked before adding the value here

**Comment by etj** [comment]
Fixed and squashed in 18cecdd76a3050df0481cfce10130c800a70bf2b

**Pull request closed without merging**